File: app/utils/responses.py
from flask import jsonify, make_response

def success_response(data=None, message="Success", status_code=200):
    """Creates a standard success JSON response."""
    response_object = {
        'status': 'success',
        'message': message
    }
    if data is not None:
        response_object['data'] = data
    return make_response(jsonify(response_object), status_code)

def error_response(message, status_code):
    """Creates a standard error JSON response."""
    response_object = {
        'status': 'error',
        'message': message
    }
    return make_response(jsonify(response_object), status_code) 

File: ui/src/services/Responses.ts
export interface SuccessResponse<T = any> {
    status: 'success';
    message: string;
    data: T;
}

export interface ErrorResponse {
    status: 'error';
    message: string;
}

export interface CredentialData {
    id: number;
    name: string;
    username: string;
    password: string;
    url?: string;
    notes?: string;
    created_at: string;
    updated_at: string;
}

export interface CredentialsResponse extends SuccessResponse<CredentialData[]> {}

export interface ExportData {
    file_url: string;
}

export interface ExportResponse extends SuccessResponse<ExportData> {}

export interface SecurityData {
    password_strength: number;
    password_breached: boolean;
}

export interface SecurityResponse extends SuccessResponse<SecurityData> {}
